src/Raven.Studio/typescript/viewmodels/resources/resources.ts

```diff
@@ -88,21 +90,109 @@ class resources extends viewModelBase {
 
         // we can't use createNotifications here, as it is called after *resource changes API* is connected, but user
         // can enter this view and never select resource
-        this.addNotification(this.changesContext.globalChangesApi().watchItemsStartingWith("db/", e => this.fetchResources()));
+        this.addNotification(this.changesContext.globalChangesApi().watchItemsStartingWith("db/", (e: globalAlertNotification) => this.fetchResource(e)));
+        //TODO: add notification for fs, cs, ts
 
         return this.fetchResources();
     }
 
+    attached() {
+        super.attached();
+        this.updateHelpLink("Z8DC3Q");
+        ko.postbox.publish("SetRawJSONUrl", appUrl.forDatabasesRawData());
+    }
+
+    // Fetch all resources info
     private fetchResources(): JQueryPromise<resourcesInfo> {
         return new getResourcesCommand()
             .execute()
             .done(info => this.resources(info));
     }
 
-    attached() {
-        super.attached();
-        this.updateHelpLink("Z8DC3Q");
-        ko.postbox.publish("SetRawJSONUrl", appUrl.forDatabasesRawData());
+    // Fetch single resource info
+    private fetchResource(e: globalAlertNotification) {
+
+        // First find if database already exists in the page resources
+        let resource = ko.utils.arrayFirst(this.resources().sortedResources(), (rs: resourceInfo) => {
+            return rs.qualifiedName === e.Id;
+        });
+
+        switch (e.Operation) {
+            case "Write": 
+                if (resource) {
+                    // Relevant for disable/enable...
+                    this.getResourceInfo(resource);
+                } else {
+                    // Add a new resource for a newly created database
+                    let newResourceInfo = this.getNewResource(e.Id.substr(0, 2), e.Id.substr(3));
+                    newResourceInfo.backupEnabled(false);
+                    this.resources().addResource(newResourceInfo); 
+                }
+               
+                break;
+            case "Load":
+                // Database turned ONLINE, get its stats
+                this.getResourceInfo(resource);
+                break;
+            case "Delete":
+                // Delete database from page resources if exists (because maybe another client did the delete..)
+                if (resource) {
+                    this.removeResource(resource);
+                }
+                break;
+        }
+    }
+
+    private getNewResource(resourceType: string, resourceName: string): resourceInfo {
```

Comment:
1. When creating new db, the server first sends a 'Write' and then 'Loaded' after a few milliseconds
2. The '...Humane' size property that is sent with 'Write' is '-1 Bytes'
3. The initial implementation thought was to show the new db in the UI  right after the the 'Write',
    and not wait for the 'Loaded' ws msg, 
    And to do so I created a new object with some default values
4. That is why you see the new resource creation and the string comparison because I did not want to show (-1) in the UI
5.  So - If we don't care about showing the new db in UI right after the 'Write'
    I will change and show only after 'Loaded' 
    And that will resolve All the other comments in the PR